### ISSUE TITLE
Add a way to browse and include uploaded images

### ISSUE THREAD

**Issue by phae** [codeforamerica/ceviche-cms#76]
I think @migurski already added a magic short-code thing for images, but I can't remember what it is, so no way anyone else ever will.  Need to show it in the UI somewhere and probably let someone see what's in an image dir.

**Comment by migurski** [comment]
There’s a working upload button for any file type, but a media concept would be a better way to do this.

**Comment by phae** [comment]
Yeah, right - isn't there also like a short code thing? {{ something.jpg }}
style I'm talking about that as well and making that more discoverable.

**Comment by phae** [comment]
Also, might be good to have an image uploader actually available on the edit page, rather than just in the file explorer.

**Comment by phae** [comment]
...and PDFs D:

**Comment by phae** [comment]
Making this an umbrella ticket. We'll scope this to actual work and such on Monday and focus just on images for now.

**Comment by phae** [comment]
To do list:
[ ] #448 Merge client and server side markdown renderer

**Comment by norrishung** [comment]
Including my notes and thoughts in here:

## Prerequisites
Merging client side and server side markdown renderer

## Potential Requirements
[1] Image Upload Form
Image Drag and Drop
[1] Caption/Alt Text for Image
[3] Image Browser
Image upload progress
Preview of what was uploaded
Image sizing guidelines
[2] Server-side or compile-time resizing
Representation in markdown
	 - Custom HTML for image? (<figure><img src="image.jpg"><figcaption>image caption</figcaption></figure>)
         - This will be important if we want to be able give them float and sizing classes.
Abstraction of image path
Delete images from library

## Implementation
Build our own
Using an existing image uploading service
  - https://uploadcare.com: IE10 plus and partial support for IE 8 and IE 9
	  - Upload widget, storage, CDN, image resizing API
 - https://www.filepicker.com: IE8 up
	 - Uploader, CDN, 
  - dropzone.js: IE10 plus and partial support for IE 8 and IE 9
  	- drag and drop widget, no server side implementation
  - https://blueimp.github.io/jQuery-File-Upload/
  
  
## Questions
Is free form images a necessary feature for any content types?
Could this just be part of a template (i.e., only allow one header image for press releases)
Should PDF and image upload be combined?
What other 'things' might need to be uploaded?
	- data, video, audio, pdf, images
  

## Notes
Will be very hard to do this with a good UX without any form of javascript
  	- preview of image
  	- modal that doesn't yank you from the page
  	- placement of image in the right part of the document
  	- drop and drop
  	- progress of uploadl

**Comment by norrishung** [comment]
Basic Image Upload Mockup.

![image_step_1](https://cloud.githubusercontent.com/assets/606439/9209143/e9eddfca-4029-11e5-95ad-4436b89c0148.png)
![image_step_2](https://cloud.githubusercontent.com/assets/606439/9209144/e9ee8862-4029-11e5-9d76-101ade208355.png)
![image_step_3](https://cloud.githubusercontent.com/assets/606439/9209142/e9eb7820-4029-11e5-97d1-693e6d4c0c84.png)

**Comment by norrishung** [comment]
View needed for no-js image upload and markdown snippet copy and paste. Would probably be something we want in the interface anyway as a dedicated media browser.

![image_library](https://cloud.githubusercontent.com/assets/606439/9212728/6a51080c-4040-11e5-962b-deecf6db9640.png)
